### ISSUE TITLE
Add status field to API

### ISSUE THREAD

**Issue by MorrisLaw** [istio/api#2900]
We need a way to communicate to the user that the CR is misconfigured. Also, whether or not a Gateway exists in order to implement Policy Attachment correctly (see [here](https://github.com/istio/api/pull/2888#issuecomment-1673384025)).  

**Task:**
The status field needs to be added to the Istio API and in addition added to the necessary proto definitions that will end up requiring the targetRef field to satisfy the requirements for Gateway APIs Policy Attachment. Those would be `AuthorizationPolicy`, `WasmPlugin`, `ProxyConfig`, `Telemetry` and `RequestAuthentication`). We can reference the [status field in K8s API objects](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) as inspiration for how this can look.

**Comment by MorrisLaw** [comment]
Thank you @linsun @therealmitchconnors. Is this where [gateway status](https://github.com/istio/istio/blob/5e5abd2e99a18f0c41c4c44e8198a27756f4115f/pilot/pkg/config/kube/gateway/controller.go#L240) is updated?

**Comment by howardjohn** [comment]
Yes but only k8s Gateway. Istio status has been a bit different to date
